jupyter/4 - dbGap FHIR Controlled Access via RAS -  COPDGene.ipynb

```diff
@@ -0,0 +1,1058 @@
+   "metadata": {},
+   "source": [
+    "## dbGaP FHIR Notebooks - Exercise 4\n",
+    "\n",
+    "In this exercise, you will query the dbGaP study \"Genetic Epidemiology of COPD (COPDGene)\". \n",
+    "\n",
+    "The study investigates Chronic Obstructive Pulmonary Disease.\n",
+    "\n",
+    "The dbGaP id for the study is phs000179.\n",
+    "\n",
+    "Note: You can use this notebook with other dbGaP studies to which you have access. Some queries will need to be adjusted to the specifics of any study that you use.\n",
+    "\n",
+    "### Learning Objectives and Key Concepts\n",
+    "\n",
+    "- Query controlled access data using RAS\n",
+    "- Show data summaries\n",
+    "- Some understanding of resources that help understand the data in dbGaP studies\n",
+    "- Use a heatmap to give high level view of data per patient\n",
+    "\n",
+    "## Motivation/Purpose\n",
+    "To make use of convenience functions to access a FHIR server, and to understand authorization methods specific to dbGaP\n",
+    "\n",
+    "## Requisites\n",
+    "- Data Access to the COPDGene study\n",
+    " -You must have been granted access to one or more dbGaP studies.\n",
+    "- Python\n",
+    " -matplotlib\n",
+    "\n",
+    " ### Icons in this Guide\n",
+    " 📘 A link to a useful external reference related to the section the icon appears in  \n",
+    "\n",
+    " 🖐 A hands-on section where you will code something or interact with the server  "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0b587755",
+   "metadata": {},
+   "source": [
+    "For functional purposes working with a controlled access study is identical to the examples already shown. The differences are:\n",
+    "- Passing a RAS token which contains authorization for the dbGaP studies you have access to \n",
+    "- Active steps to ensure the data are used within the Data Use Conditions you have agreed to for the datasets\n",
+    "\n",
+    "FHIR security as described by HL7 deals with passing credentials in the request header. The standard RAS GA4GH Passport 1.2 standard passports are too large to be included in a header. Therefore a mechanism to obtain a smaller Task Specific Token (TST) has been created.\n",
+    "\n",
+    "📘 Notes on FHIR security\n",
+    "\n",
+    "The TST mechanism uses the central NIH RAS service for authentication, but then provides a smaller token. Use of this mechanism is not described here.\n",
+    "\n",
+    "Typically the token is delivered to the Downloads folder by the web browser a user has used to visit the TST site.\n",
+    "\n",
+    "For simplification the dbgapfhir class handles adding the token to all requests it makes.\n",
+    "\n",
+    "#TODO - a separate notebook explaining at lower level how to use the requests module to pass the TST."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "4b6595ac",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from dbgapfhir import dbgapfhir\n",
```

Comment:
In Python, the convention is to use PascalCase in class names. So, I'd name this class DbGaPFhir

Reply:
Good suggestion.